pkg/auth/asynccache.go

```diff
@@ -27,13 +31,21 @@ func NewAsyncCache[T any](ctx context.Context, reloadPeriod time.Duration, cachi
 		cachingFunc:  cachingFunc,
 	}
 
-	item, err := cachingFunc(ctx)
+	var err error
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	wait.UntilWithContext(ctx, func(ctx context.Context) {
```

Comment:
ftr - `PollUntilContextTimeout` might fit you even better. Sorry about the confusion.